### PR TITLE
Migrate command `regenerate-interleaved-storage-index-file`

### PR DESCRIPTION
**Pull request by zymap**
Descriptions of the changes in this PR:

#2034

### CODE REVIEW

**Comment by zymap** [comment]
run integration tests

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        try {
+            return generate(conf, cmdFlags);
+        } catch (Exception e) {
+            throw new UncheckedExecutionException(e.getMessage(), e);
+        }
+    }
+
+    private boolean generate(ServerConfiguration conf, RISIFFlags flags) throws NoSuchAlgorithmException, IOException {
+        byte[] password;
+        if (!flags.password.equals(DEFAULT)) {
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java]: password can be `null`, no?

I think you also need to add a test to cover if password is null.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            password = flags.password.getBytes(StandardCharsets.UTF_8);
+        } else if (!flags.b64Password.equals(DEFAULT)) {
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java]: same comment as above

**Comment by zymap** [comment]
@sijie  I resolve it by pre check. PTAL

**Comment by zymap** [comment]
run pr validation

**Comment by sijie** [comment]
@zymap there is one checkstyle violation. can you check?

```
[ERROR] /home/jenkins/jenkins-slave/workspace/bookkeeper_precommit_pullrequest_validation/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java:49:8: Unused import: java.util.Set. [UnusedImports]

```

**Comment by zymap** [comment]
run bookkeeper-server replication tests

**Comment by zymap** [comment]
run pr validation

**Comment by zymap** [comment]
run integration tests
run bookkeeper-server replication tests

**Comment by sijie** [comment]
run integration tests
run bookkeeper-server replication tests

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        ServerConfiguration serverConfiguration = new ServerConfiguration(conf);
+        InterleavedStorageRegenerateIndexOp i = new InterleavedStorageRegenerateIndexOp(serverConfiguration, ledgerIds,
+                                                                                        password);
+        i.initiate(flags.dryRun);
+
+        LOG.info("-- Done rebuilding index file for {} --", ledgerIds);
+        return true;
+    }
+
+    private void checkNPE(RISIFFlags flags) {
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java]: nit : a better name will be `validateFlags`

Comment (zymap) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java]: ok, thanks
